### DISCUSSION TITLE
Content is not found for the url of our own Maven repository

### DISCUSSION THREAD

**Discussion by taolulu** [renovatebot/renovate#29476] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Renovate encountered issues while checking our own Maven repository. It encountered a 404 error while trying to check the version of our own library, but I can access the corresponding URL normally from the browser:
```ndjson
{"name":"renovate","hostname":"runner-bb2txvos-project-859-concurrent-0","pid":18,"level":20,"logContext":"fq6FeU1Luo7uAoCPFSnSX","repository":"android/component","msg":"GET https://maven.oursource.com/repository/android/com/ourcompany/android/ourlib/maven-metadata.xml = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=230)","time":"2024-06-06T08:11:07.273Z","v":0}
{"name":"renovate","hostname":"runner-bb2txvos-project-859-concurrent-0","pid":18,"level":20,"logContext":"fq6FeU1Luo7uAoCPFSnSX","repository":"android/component","url":"https://maven.oursource.com/repository/android/com/ourcompany/android/ourlib/maven-metadata.xml","msg":"Content is not found for Maven url","time":"2024-06-06T08:11:07.273Z","v":0}
```

I am not sure what I've missed. My config is simple :
```Json
{
  "extends": [
    "config:recommended"
  ],
  "packageRules": [
    {
      // some match package patterns...
    },
    {
      "matchDatasources": ["maven"],
      "registryUrls": ["https://maven.somesource.com", "https://maven.oursource.com/repository/android/"]
    }
  ]
}

```
And the host rules for our repository defined in the gitlab ci config file:
```yaml
   RENOVATE_HOST_RULES: '[{"hostType": "maven", "matchHost": "https://maven.oursource.com","username": "${USERNAME}","     password": "${PASSWORD}"}]'
```
This setup seems to work correctly,  since the job artifact shows the right username, and Renovate applied the basic authentication before looking up our repository:
```ndjson
[{"hostType":"maven","matchHost":"https://maven.oursource.com","username":"deployer","password":"***********"}]

{"name":"renovate","hostname":"runner-bb2txvos-project-859-concurrent-0","pid":18,"level":20,"logContext":"fq6FeU1Luo7uAoCPFSnSX","repository":"android/component","msg":"hostRules: applying Basic authentication for maven.oursource.com","time":"2024-06-06T08:11:07.041Z","v":0}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
It's probably a problem with the env substitution and they are empty or undefined

**Reply by taolulu** [reply]
OK, I'll check env tomorrow

**Reply by taolulu** [reply]
Which environment variables could it be related to?

**Reply by rarkins** [reply]
The username and password

**Reply by taolulu** [reply]
I can see the right username in the job artifact, doesn't this mean that these variables have been set correctly?

**Reply by rarkins** [reply]
It doesn't necessarily mean that their value is being correctly substituted into the host rules you've defined. You need to make sure they are "substituted" before being passed to renovate.

**Reply by taolulu** [reply]
i still got no idea with this problem... So how can I verify that these values have been correctly passed to Renovate?
